### DISCUSSION TITLE
Can Tasmota send MQTT messages only on teleperiod?

### DISCUSSION THREAD

**Discussion by Bebuccio** [arendst/Tasmota#18135] [Q&A]
I don't know if is possible limit the MQTT messages only at TelePeriod time. I want to disable MQTT messages when commands are executed (i.e. When I use a WebSend command, also during a rule, I see the MQTT message "13:00:03.182 MQT: stat/MyTamotaBoard/RESULT = {"WebSend":"Done"}")
More generic, it is possible to filter which topics can be published?
Thanks

**Comment by barbudor** [comment]
No. There is no feature for filtering what is sent.

Why is that a problem?
We have users with 200+ devices that are fine with how it works

**Reply by Bebuccio** [reply]
I have connected the Tasmota directly to AWS MQTT server (I don't use a private MQTT-Broker). When the Tasmota sends messages to AWS, there is a cost on the Amazon Service. I want to limitate this traffic. If there is an Option to limit ONLY on the "tele/xxx/yyy" messages, or only when there is an explicit publish-command, it will be cheeper the use of AWS service filtering the unnecessary traffic.

**Reply by s-hadinger** [reply]
Cost is $1.00 per million MQTT messages. How many devices do you have? Unless it's a very high number it will be only cents

**Reply by Bebuccio** [reply]
I have no much devices but they have rules that periodically do some commands. For each commands Tasmota publish a MQTT message for every (unusefull) JSon-RESULT.
So calculating the messages that can be generated per day, I have a lot of MQTT publish with no utility.
If there was a tasmota-command that can, or filter the payload, or stops the RESULT payload, this can be usefull also to reduce internet traffic (in this moment the internet traffic to reach AWS is over a mobile-network).
I also saw that if I temporarily disable the MQTT, when I re-enable it the Tasmota-Device reboots.

**Reply by s-hadinger** [reply]
Can you give numbers? I would like to sense what is "a lot of MQTT publish".

**Reply by sfromis** [reply]
If you have automations with rules firing off lots of commands, in many cases you can tweak the logic to have far fewer rule triggers firing, with associated commands. One example is to use the one-shot option to prevent triggers from re-firing while the condition is still true.
